fhir-model/src/test/java/org/linuxforhealth/fhir/model/string/util/test/StringSizeControlStrategyTest.java

```diff
@@ -0,0 +1,140 @@
+     * to truncate the unicode character since max allowed bytes = 4.
+     * @throws FHIROperationException
+     */
+    @Test
+    public void testStringSizeControlStrategy_WithInpEqSizeLimit() throws FHIROperationException {
+        StringSizeControlStrategyFactory factory = StringSizeControlStrategyFactory.factory();
+        assertNotNull(factory);
+        org.linuxforhealth.fhir.model.string.util.strategy.StringSizeControlStrategy strategy = factory.getStrategy(Strategy.MAX_BYTES.getValue());
+        String description = "ab\u0080"; // padding character
+        int maxBytes = 4;
+        String truncatedDescription = strategy.truncateString(description, maxBytes);
+        assertTrue(truncatedDescription.equals(description));
+        assertTrue(truncatedDescription.getBytes().length == maxBytes);
+        assertTrue(truncatedDescription.getBytes().length == description.getBytes().length);
+    }
+    
+    /**
+     * Test an input string which contains a unicode character which occupies 2 bytes of memory. The expected output is 
+     * to truncate the unicode character since max allowed bytes = 3.
+     * @throws FHIROperationException
+     */
+    @Test
+    public void testStringSizeControlStrategy_WithUnicodeInpExceedingSizeLimit() throws FHIROperationException {
+        StringSizeControlStrategyFactory factory = StringSizeControlStrategyFactory.factory();
+        assertNotNull(factory);
+        org.linuxforhealth.fhir.model.string.util.strategy.StringSizeControlStrategy strategy = factory.getStrategy(Strategy.MAX_BYTES.getValue());
+        String description = "ab\u0080"; // padding character
+        int maxBytes = 3;
+        String truncatedDescription = strategy.truncateString(description, maxBytes);
+        assertFalse(truncatedDescription.equals(description));
+        assertTrue(truncatedDescription.getBytes().length <= maxBytes);
+        assertFalse(truncatedDescription.getBytes().length == description.getBytes().length);
+    }
+    
+    /**
+     * Test an input string which contains a surrogate pair which occupies 4 bytes of memory. The expected output is 
+     * to truncate the unicode character since max allowed bytes = 4.
+     * @throws FHIROperationException
+     */
+    @Test
+    public void testStringSizeControlStrategy_WithSurrogatePairInpEqSizeLimit() throws FHIROperationException {
+        StringSizeControlStrategyFactory factory = StringSizeControlStrategyFactory.factory();
+        assertNotNull(factory);
+        org.linuxforhealth.fhir.model.string.util.strategy.StringSizeControlStrategy strategy = factory.getStrategy(Strategy.MAX_BYTES.getValue());
+        String description = "\uD834\uDD1E"; // surrogate pair
+        int maxBytes = 4;
+        String truncatedDescription = strategy.truncateString(description, maxBytes);
+        assertTrue(truncatedDescription.equals(description));
+        assertTrue(truncatedDescription.getBytes().length == maxBytes);
```

Comment:
whenever you use getBytes you should specify the character coding (which in our case should always be UTF-8, like this `truncatedDescription.getBytes(StandardCharsets.UTF_8);`